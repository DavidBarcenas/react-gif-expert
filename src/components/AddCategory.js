import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [value, setValue] = useState('');

  const handleValueChange = (e) => setValue(e.target.value);
  const handlesubmit = (e) => {
    e.preventDefault();
    if (value.trim().length > 2) {
      // setCategories((cats) => [...cats, value]);
      setCategories((cats) => [value]);
      setValue('');
    }
  };
  return (
    <form onSubmit={handlesubmit}>
      <input
        type="text"
        placeholder="Search..."
        value={value}
        onChange={handleValueChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
