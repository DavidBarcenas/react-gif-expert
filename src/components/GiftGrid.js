import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GiftGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <div>
      <h3 className="category__subtitle">{category}</h3>
      {loading && <p>Loading...</p>}
      <ul className="Gifs__list fadeInUp">
        {images.map((img) => (
          <li key={img.id}>
            <img src={img.url} alt={img.title} />
          </li>
        ))}
      </ul>
    </div>
  );
};
