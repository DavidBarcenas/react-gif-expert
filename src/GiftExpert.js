import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GiftGrid } from './components/GiftGrid';

const GiftExpert = ({ dafultCategory = [] }) => {
  const [categories, setCategories] = useState(dafultCategory);

  // const handleAdd = () => {
  //   setCategories([...categories, 'Batman']);
  // };

  return (
    <div>
      <h2>GiftExpert</h2>

      <AddCategory setCategories={setCategories} />

      <hr />

      {categories.map((c) => (
        <GiftGrid key={c} category={c} />
      ))}
    </div>
  );
};

export default GiftExpert;
