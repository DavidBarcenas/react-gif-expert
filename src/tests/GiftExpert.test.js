import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GiftExpert from '../GiftExpert';
import { GiftGrid } from '../components/GiftGrid';

describe('<GifExpert /> testing', () => {
  test('should render correctly', () => {
    const wrapper = shallow(<GiftExpert />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show a list of categories', () => {
    const categories = ['SpiderMan', 'Iron-Man'];
    const wrapper = shallow(<GiftExpert dafultCategory={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GiftGrid').length).toBe(categories.length);
  });
});
