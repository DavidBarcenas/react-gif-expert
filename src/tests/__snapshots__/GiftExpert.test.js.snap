// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpert /> testing should render correctly 1`] = `
<div>
  <h2>
    GiftExpert
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</div>
`;

exports[`<GifExpert /> testing should show a list of categories 1`] = `
<div>
  <h2>
    GiftExpert
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GiftGrid
    category="SpiderMan"
    key="SpiderMan"
  />
  <GiftGrid
    category="Iron-Man"
    key="Iron-Man"
  />
</div>
`;
